### DISCUSSION TITLE
new-target-typed and switch pattern expression

### DISCUSSION THREAD

**Discussion by FaustVX** [dotnet/csharplang#4960] [Q&A] [answered]
I have this situation:
(Using .NET 5, C# 9)
```csharp
public abstract class History {
  public class Single : History { … }
  public class Multi : History {
    public List<Single> Positions { get; }
  }

  public static Multi Add(History last, Single next)
    => last switch {
      Single s // IDE0059 : Unnecessary assignment of a value to 's'
        => new() { // but : new Multi() here works well
          Positions = {
           s, // CS0103 : The name 's' does not exist in the current context
           next,
          }
        },
      Multi m => …
    };
}
```
[Sharplab.io](https://sharplab.io/#v2:C4LgbgpgTglgZgTwDQBMQGoA+ABADAAmwEYAWAbgFgAobAZkICZ8BhfAb2vy8PuxPwCyACgCU7TtwC+ErnXwBDAEYBnYFHkBjYI3wAJGKoD2UBOKqzeTAMowAdgHMANhHwg9B4MdNt808zx0BAFdHYBhXdyMTM24AgBkPAB4bB2cAPnwABUNlGDDDW2V2fHsIYDJfCT8JOWIANkEQsPwAQRQUZkd5ZVyNeUchfSjTLtUkfBSnF1sIAA9gERl8AF4M0e1lAHc8jQALGNiJuyn8ZSXY1fwZzcbQmFFigHpHq4gb4LuH3egXTeMAayKmwgjkc50O2Vy+UKKwOh1iyiQ4MOM3mSP88K4fnhknR8I+zQAtisMtdRHipJQqDVeEQGgTwm0UAAVeRQUrAZkIAAOEBQgw8Xnw63Gk2cr3miwxJOF3Q222Aezh3DFLiKz3wAEkACIAUVwuAArABOCIAVVsMw0EB6bNM3Vy9lshIgtm0hjgCnwYH6QRcnnwAHJlIHkTKyWIfBrrvgvj98ChDDbbIHtH8oICw1xIXkYAUistlZjEfgNcwrLgiLh6G5md8rvIXUGQwmk0VbIZtHMPPg7PhgPWNEEoFBXdoNAVgHNgFmuKjgBTMZVpVjF9wGfhiZcI2vJFS/JIgA=)
Just replacing the `new()` with classical `new Multi()` suppress the warning and error.

**Comment by alrz** [comment, marked as answer]
Duplicate of https://github.com/dotnet/roslyn/issues/54193
